benchmark_runner/common/template_operations/template_operations.py

```diff
@@ -94,12 +94,15 @@ def __build_template_data(self, previous_data: dict, template_root_data: dict):
                 **extra_data, **extra_kind_data, **extra_runtype_data, **extra_kind_runtype_data}
 
     @logger_time_stamp
-    def generate_yamls_internal(self, scale: str = None, scale_node: str = None, redis: str = None):
+    def generate_yamls_internal(self, scale: str = None, scale_num: str = None, scale_node: str = None, redis: str = None):
```

Comment:
It looks like `scale` here is really an index, and `scale_num` is the total scale.  I suggest naming these parameters `scale_idx` and `scale` or similar.

These should be passed in as ints, and this routine should convert them to strings.  These are conceptually numbers, but they need to be converted into strings for template purposes.  That should be done below.

Reply:
Casts should happen as close to the point of use as possible.  The caller shouldn't know the use to which the data is being put.  What if the callee needs to use it both ways?

Reply:
In this way, I cast it once and not several times

Reply:
Why does that matter?  This is not extremely performance-sensitive code where the cast value needs to be cached.

Reply:
If you need to cache something for performance reasons, that's one thing.  If not, it's a lot easier to read and maintain if it's passed around in its conceptual form and only cast at point of use.